src/System.Reflection.Metadata/src/System/Reflection/BlobWriter.cs

```diff
@@ -86,7 +86,7 @@ public int Offset
             {
                 if (value < 0 || _start > _end - value)
                 {
-                    ValueArgumentOutOfRange();
+                    throw new ArgumentOutOfRangeException(nameof(Offset));
```

Comment:
Please revert this.  Adding throws into such a setter can affect inlining, making this a change that could have a potential impact on performance.